src/boot/bootctl.c

```diff
@@ -842,7 +853,15 @@ static int copy_one_file(const char *esp_path, const char *name, bool force) {
         if (s)
                 *s = 0;
 
-        p = strjoina(BOOTLIBDIR "/", name);
+        r = chase_symlinks(BOOTLIBDIR "/", arg_root, CHASE_PREFIX_ROOT | CHASE_TRAIL_SLASH, &path, NULL);
+        if (r < 0)
+                return log_error_errno(r,
+                                       "Failed to resolve path %s%s%s: %m",
+                                       BOOTLIBDIR,
+                                       arg_root ? " under directory " : "",
+                                       arg_root ?: "");
+
+        p = strjoina(path, name);
```

Comment:
path_join() sounds like the better choice here. then you can drop the "/" suffixing above and the CHASE_TRAI_SLASH, and the (unlikely) special case that BOOTLIBDIR might be the root dir will be covered nicely...

Reply:
pre-existing conditions but ok, updated, ptal